(content): Add darker mood Coalition culture conversations (#9790)
endless-sky/endless-sky: data/coalition/coalition culture conversations.txt

				decline

mission "Coalition Departing Trouble - A"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 2
	source
		government "Coalition"
		not attributes "coalition station"

> 	destination "Ring of Friendship"

Reviewer: Would it make sense to fail these missions after the `on enter`? As I understand it, they would be in your save file until you land on the ring, even though they don't really do anything anymore... (I might be mistaken, though.)

Author: Removed destination and added that.

	on enter
		conversation
			`Your ship narrowly avoids a few small Coalition vessels as you exit <origin>'s atmosphere. Not a second later, while you're still stunned by the near-crash, once again the <ship> is almost hit by incoming ships, this time Heliarch warships, in what looks like an intense chase after the ships from before.`

> 			`	After they've all passed, you check your comms and radar, finding some other civilian ships nearby seem to have experienced a similar shock of their own. Some of the ships linger over <origin> for a while, as if unsure of what to do. Others bolt away with engines at full power, leaving the system in a blink, and some eventually decide to land again, either too curious about the situation, or perhaps wanting to avoid any trouble.`

Reviewer: ```suggestion
			`	After they've all passed, you check your comms and radar, finding some other civilian ships nearby that seem to have experienced a similar shock of their own. Some of the ships linger over <origin> for a while, as if unsure of what to do. Others bolt away with engines at full power, leaving the system in a blink, and some eventually decide to land again, either too curious about the situation, or perhaps wanting to avoid any trouble.`
```
The original sentence is fine, but I think this is more understandable.

mission "Coalition Departing Trouble - B"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 2
	source
		government "Coalition"
	destination "Ring of Friendship"
	on enter
		conversation
			`Once you're a good distance from <origin>, your sensors alert you to danger in the system: several explosions and a torrent of projectiles are flying nearby. You swiftly ready yourself and prepare for the threat, only to find Coalition civilian ships blowing up near the middle of the system. Some Heliarch warships are in their midst, moving from one to the other, unbothered by the ship debris as vessels explode around them.`
			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`

mission "Coalition Departing Trouble - C"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		not "assisting lunarium"
		not "joined the heliarchs"
		random < 2
	source
		government "Coalition"
	destination "Ring of Friendship"
	on enter
		conversation

> 			`You depart to find something of a screen of Heliarch warships around <origin>, circling it and approaching all departing civilian ships. It's no different with your <ship>, and soon enough you're hailed by a Heliarch Punisher approaching your position.`

Reviewer: ```suggestion
			`You depart to find something like a screen of Heliarch warships around <origin>, circling it and approaching all departing civilian ships. It's no different with your <ship>, and soon enough you're hailed by a Heliarch Punisher approaching your position.`
```

			`	"Salutations, Captain <last>. Performing a routine inspection, we are. Board you momentarily we shall, to verify the safety of your vessel, then leave you to your business we shall."`

> 			`	They sign off before you can respond, and the Punisher readies to board the <ship>, as more warships gather around the scene. Dozens of armed Heliarch agents march into your ship silently, diligently going about checking the bunks, cockpit, engine room...`

Reviewer: It might be cool to spawn some NPCs so the player can actually see all the heliarch ships around the planet. Honestly, some of the other ones could use NPCs as well, but I'm not sure if it's worth all the trouble.

mission "Coalition Incident: Secret Sky"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 3
	source "Secret Sky"
	on offer
		conversation
			`You see an Arach captain crossing the spaceport again and again, to reach the few people in sight here. He converses with each one briefly, showing something in a small monitor, and then moving on to the next shortly after. After a while, he gets to you.`
			`	"Captain? A moment if free you are?" He asks scrambling his legs to light up the monitor. It displays a picture of a young Kimek. "A passenger of mine. Stopped by here to drop a few Saryd pilgrims, we had, but after seen them off I had, find this Kimek back in my ship, I couldn't!" He exclaims, pointing to an Arach Transport parked just before the edge of where <origin>'s dense fog cuts off your view. "Seen him anywhere, have you?"`
			choice
				`	"Sorry, I haven't. Did you try getting in touch with the local Heliarchs?"`
				`	"Sorry, but I haven't seen him either."`
					goto end

> 			`	He nods. "Contacted them a few hours ago, I had. Look for my passenger in the restricted sections, I wished to, but allowed there I am not, so figured that ask them for aid, I could." He sighs. "Filed a report and sent a few agents off to look for him, they did, but so far received I have not any word from them."`

Reviewer: ```suggestion
			`	He nods. "Contacted them a few hours ago, I had. Look for my passenger in the restricted sections, I wished to, but allowed there I am not, so figured that ask them for aid, I could." He sighs. "Filed a report and sent a few agents off to look for him, they did, but so far received I have not any word from them."`
```
Is the end of the last sentence properly coalition-ed? I might just be confusing myself tbh, not sure.

Author: Changed it to what I think is more common Coal speech.

		government "Coalition"
		attributes "kimek"
		not attributes "coalition station"
		not attributes "farming"
	on offer
		conversation
			`The spaceport here on <origin> is in something of an uproar. Thousands of Kimek are seemingly organized in what, strangely enough for a world in the Coalition, resembles a protest. Carrying about some signs or devices repeating a message in the Kimek language, or an icon that depicts the usual Kimek diet, with their many crops sharing some space with a few fish. Strangest of all, however, is how the local Heliarchs are handling the situation: they appear almost hesitant, defensively speaking to the civilians in attempts to calm them down, sometimes apologizing or simply being forced to listen to the complaints as several Kimek pile up around them.`
			`	When you finally manage to find a Heliarch who isn't occupied trying to keep the area in order, a Saryd, he explains the situation. "Missed its delivery, a food convoy has. Still investigating the issue, we are; know only that, somehow, gone missing, the entire fleet has..." He sighs, shaking his head. "Keep the locals calm we must." He goes on to help the other Heliarchs, without another word.`

> 			`	Within a few minutes that must've felt like hours for the stationed Heliarch agents, battalions of drones start making their way from the tallest building in the city - the hunger tower. Ferrying around several meals each, with impressive precision, the drones fly throughout the civilian mass, distributing the meals. Hundreds of larger head to the residential areas of the city, presumably to provide the necessary supplies for the citizens until the next food convoy arrives. Seemingly satisfied, for the moment at least, the Kimek gradually dissipate out of the spaceport and back to their homes, leaving the spaceport to the now exhausted Heliarch agents.`

Reviewer: ```suggestion
			`	Within a few minutes that must've felt like hours for the stationed Heliarch agents, battalions of drones start making their way from the tallest building in the city - the hunger tower. Ferrying around several meals each, with impressive precision, the drones fly throughout the civilian mass, distributing the meals. Hundreds of larger ones head to the residential areas of the city, presumably to provide the necessary supplies for the citizens until the next food convoy arrives. Seemingly satisfied, for the moment at least, the Kimek gradually dissipate out of the spaceport and back to their homes, leaving the spaceport to the now exhausted Heliarch agents.`
```

				decline

mission "Coalition Incident: Arach Store"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 2
	source
		government "Coalition"
		attributes "arach"
		not attributes "coalition station"
		not planet "Ahr"
	on offer
		conversation
			`A Heliarch patrol seems to be discussing with an Arach shop worker in front of a large, prestigious looking Arach clothing store in the spaceport. Other agents prevent you and other bystanders from approaching the scene, but you can still make out some context from their gesturing. It seems the Heliarchs are there to confiscate some pieces from the shop that you guess don't abide by their laws, while the shop worker keeps worryingly attempting to answer what questions they pose him.`

> 			`	Eventually, the shop owner arrives. Judging by her fine clothes, and from the reactions of the local Arachi, you figure she must either be really famous here on <origin>, or possibly belong to one of the Arach Houses, though nothing on her attire seems to indicate any specific house. She tranquilizes the worker, and takes on to speaking with the Heliarch agents as he heads back inside.`

Reviewer: > She tranquilizes the worker

Uhh... what? The boss just walks up to him and shoots/drugs him?

Author: It would appear that is not a very good direct translation from portuguese. Whoopsie. Changed it.

			`	She tries to diffuse the situation, speaking calmly and attempting to take the items back into the shop, but the Heliarch agents do not budge. Running out of patience, the Heliarch in charge of the patrol cuts the Arach lady off as she's speaking, forcibly taking all the items in question, and having the agents rip apart what you guess must've been the offending parts of the cloth pieces. Stunned, the Arach simply watches as they render most of the items into unusable rags, keeping the bits they hacked away, and handing her back the other tattered remains. Without another word, they leave the scene. The Arach heads on inside, and closes the store.`
				decline

mission "Coalition Incident: Ring of Wisdom"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"

> 		random < 1

Reviewer 2: Cant we get it higher since it only offers on the ring of wisdom? (Say 2-4%, maybe?)
Or mb offer it at the other rings, too?

Reviewer 3: Also the Ring of Wisdom is the one world in Coalition space the player will visit the most before choosing a side in the conflict, so even 1% is still likely to be seen by players.

		conversation
			`A few minutes into your walk around the spaceport of the <origin>, suddenly all lighting shifts from the usual, comfortable tones you're used to, to a cacophony of flashing warnings and warning messages throughout all sorts of screens, amidst unravelling siren blasts that repeat messages in all three Coalition languages in a cycle. You look around, and not only are the civilians in the ring also alarmed as you are, all Heliarch agents begin moving in a hurried, albeit organized pace, towards restricted sections, Heliarch-only rooms and warehouses.`
			`	It doesn't take long before hundreds of other Heliarchs come pouring in from other sections of the ring, and when the flux finally stops, you see them take control of the machinery near the edge of the ring section, bringing down an immense wall, akin to a storefront gate, and sealing the section off from the rest of the ring. Ship traffic is also cut off entirely, with Heliarch squadrons keeping any ship from departing, and warships visibly walling off the hangar entrances. The Heliarchs organize the civilians swiftly, instructing them to stay far away from the wall toward the outer face of the ring section.`

> 			`	As you too are being herded by the agents closer to the opposing walls, you manage to get some context out of one of the agents - a Heliarch team that was exploring one of the ring's unknown sections had some sort of accident, and one of the members was tracked as coming to this section. Heliarch agents stand armed and ready, pointing their weapons to the entrance to the restricted sections, and so as if rendered mute by anticipation, all in the quarantined ring section appear to hold their breath, waiting.`

Reviewer: ```suggestion
			`	As you too are being herded by the agents closer to the opposing walls, you manage to get some context out of one of the agents - a Heliarch team that was exploring one of the ring's unknown sections had some sort of accident, and one of the members was tracked as coming to this section. Heliarch agents stand armed and ready, pointing their weapons towards the entrance of the restricted sections, and as if rendered mute by anticipation, all in the quarantined ring section appear to hold their breath, waiting.`
```

			`	Sure enough, the many minutes of silence are disrupted by echoing, increasingly louder screaming - one scream, incessant, coming from the shut passageway watched over by the sights of what must be a hundred Heliarch guns. A thud at the shut door briefly peaks out amidst the continuous shout. It opens to a lone, frantic Saryd, with a deeply perturbed look in his eyes. He doesn't stop at the sight of the several gun barrels pointed at him, instead drawing his own weapon, only to point it back to where he came from, firing wildly at the dark passage.`
			`	Then it stops. The screaming, and the gunshots, both stop abruptly, with the Saryd dropping the gun, then following it to the ground. He lays there, unresponsive, catatonic, as the Heliarch agents head closer to inspect him, eventually blocking him out entirely from view. A few seconds later, some of the Heliarchs back up, then bolt to a comms room. Others stand firmly on guard, with their guns pointed either at the Saryd, or at the passageway from which he came.`

> 			`	Eventually, a team comes with what vaguely resembles a very large gurney, and some sort of large sack. The Heliarch agents surrounding the Saryd shuffle out of the way for those carrying the equipment to pass, and within a few minutes they're bringing the Saryd away, encased in the sack as if in a body bag, carried on the gurney. Halfway to the shut down section doors, they stop as the Saryd moves erratically inside the sack, kicking and flailing wildly. It stops within a few seconds, and they resume their way out of the quarantined ring portion.`

Reviewer: ```suggestion
			`	Eventually, a team comes with what vaguely resembles a very large gurney, and some sort of large sack. The Heliarch agents surrounding the Saryd shuffle out of the way for those carrying the equipment to pass, and within a few minutes they're taking the Saryd away, encased in the sack as if in a body bag, carried on the gurney. Halfway to the shut down section doors, they stop as the Saryd moves erratically inside the sack, kicking and flailing wildly. It stops within a few seconds, and they resume their way out of the quarantined ring portion.`
```

				decline
			label lunarium
			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet. They pay you no mind, and much to your relief steer a safe distance away from your ship this time. You cannot help but wonder whether the ships they were chasing held any Lunarium members, or if they were simply more unfortunate civilians.`
	npc
		personality launching fleeing uninterested
		government "Heliarch"
		fleet
			names "heliarch"
			variant
				"Heliarch Hunter"
				"Heliarch Interdictor"
				"Heliarch Breacher"

mission "Coalition Departing Trouble - B"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 2
	source
		government "Coalition"
	on enter
		fail
		conversation
			`Once you're a good distance from <origin>, your sensors alert you to danger in the system: several explosions and a torrent of projectiles are flying nearby. You swiftly ready yourself and prepare for the threat, only to find Coalition civilian ships blowing up near the middle of the system. Some Heliarch warships are in their midst, moving from one to the other, unbothered by the ship debris as vessels explode around them.`

> 			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`
> 			branch heliarch
> 				has "joined the heliarchs"
> 			branch lunarium
> 				has "joined the lunarium"
> 			branch neither
> 				not "joined the heliarchs"
> 				not "joined the lunarium"
> 			label heliarch
> 			`	One of the ships hails you briefly, apparently in a cheery mood as they celebrate having captured the criminals. They wish you a good day and good luck on your duties, and cut the comms as they prepare to leave.`
> 				decline
> 			label lunarium
> 			`	As you glance at the wreckage of the civilian ships, you feel uneasy. Worse still is the thought that hits you soon after, of whether any of the Lunarium leaders you met were aboard any of the ships.`
> 				decline
> 			label neither
> 				decline

Reviewer 4: It does work as Arachi says, but it usually adds an empty line between the last paragraph and the `[done]` button. That could be avoided by doing this:
```suggestion
			branch heliarch
				has "joined the heliarchs"
			branch lunarium
				has "joined the lunarium"
			branch neither
				not "joined the heliarchs"
				not "joined the lunarium"
			label heliarch
			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`
			`	One of the ships hails you briefly, apparently in a cheery mood as they celebrate having captured the criminals. They wish you a good day and good luck on your duties, and cut the comms as they prepare to leave.`
				decline
			label lunarium
			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`
			`	As you glance at the wreckage of the civilian ships, you feel uneasy. Worse still is the thought that hits you soon after, of whether any of the Lunarium leaders you met were aboard any of the ships.`
				decline
			label neither
			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`
				decline
```

			`	"Salutations, Captain <last>. Performing a routine inspection, we are. Board you momentarily we shall, to verify the safety of your vessel, then leave you to your business we shall."`
			branch lunarium
				has "joined the lunarium"
			`	They sign off before you can respond, and the Punisher readies to board the <ship>, as more warships gather around the scene. Dozens of armed Heliarch agents march into your ship silently, diligently going about checking the bunks, cockpit, engine room...`
			`	Within a few minutes, they appear satisfied with turning your ship upside down, and all march back to the Punisher. They thank you for "cooperating", and the ships clear the way for you to move out.`
				decline
			label lunarium
			`	They sign off before you can respond, and the Punisher readies to board the <ship>, as more warships gather around the scene. Although you're not currently performing any tasks for the Lunarium, you still grow increasingly worried as the Punisher approaches and connects to your ship. You do your best to keep a calm profile, as dozens of armed Heliarch agents march into your ship silently, diligently going about checking the bunks, cockpit, engine room...`

> 			`	Within a few minutes, they appear satisfied with turning your ship upside down, and all march back to the Punisher. They thank you for "cooperating", and the ships clear the way for you to move out. Once they're well enough away from the <ship>, you take to scouring it for anything they might've left behind to keep tabs within your ship. Much later on in the day, after you've confirmed to have found nothing of the sort, you breath a sigh of relief.`

Reviewer 5: ```suggestion
			`	Within a few minutes, they appear satisfied with turning your ship upside down, and all march back to the Punisher. They thank you for "cooperating", and the ships clear the way for you to move out. Once they're well enough away from the <ship>, you take to scouring it for anything they might've left behind to keep tabs within your ship. Much later on in the day, after you've confirmed to have found nothing of the sort, you breathe a sigh of relief.`
```

				decline

mission "Coalition Incident: Ring of Wisdom"
	minor
	invisible
	to offer
		has "Coalition: First Contact: done"
		random < 1
	source "Ring of Wisdom"
	on offer
		conversation
			`A few minutes into your walk around the spaceport of the <origin>, the lighting suddenly shifts from the usual, comfortable tones to a cacophony of flashing warnings. Diagnostic warning messages display on all sorts of screens, and siren blasts are interrupted only by alarm messages in all three Coalition languages on repeat. You notice that not only are the civilians in the ring as alarmed as you, but every Heliarch agent you can see begins moving in a hurried, albeit organized pace, towards restricted sections, Heliarch-only rooms and warehouses.`

> 			`	It doesn't take long before hundreds of other Heliarchs come pouring in from other sections of the ring. When the stream finally stops, you see them take control of the machinery near the edge of the ring section, bringing down an immense wall, akin to a storefront gate, and sealing the section off from the rest of the ring. Ship traffic is also cut off entirely, with Heliarch squadrons keeping any ship from departing, and warships visibly walling off the hangar entrances. The Heliarchs organize the civilians swiftly, instructing them to stay far away from the wall toward the outer face of the ring section.`

Reviewer 6: ```suggestion
			`	It doesn't take long before hundreds of other Heliarchs come pouring in from other sections of the ring. When the stream finally stops, you see them take control of the machinery near the edge of the ring section, activating a huge retractable barrier that seals this section off from the rest of the ring. Ship traffic is also cut off entirely, with Heliarch squadrons keeping any ship from departing, and warships visibly walling off the hangar entrances. The Heliarchs organize the civilians swiftly, moving them away from an area where most of the agents are gathering.`
```

> 			`	As you too are herded by the agents closer to the opposing walls, you manage to get some context out of one of the agents - a Heliarch team that was exploring one of the ring's unknown sections had some sort of accident, and one of the members was tracked as coming to this section. Heliarch agents stand armed and ready, pointing their weapons towards the entrance of the restricted sections, and as if rendered mute by anticipation, all in the quarantined ring section appear to hold their breath, waiting.`

Reviewer 6: ```suggestion
			`	As you too are herded away by the agents, you manage to get some context out of one of them - a Heliarch team that was exploring one of the ring's unknown sections had some sort of accident, and one of the members is expected to return at the area where most of the agents are gathering. The Heliarchs stand armed and ready, pointing their weapons toward a door leading to the restricted sections, and as if rendered mute by anticipation, all in the quarantined ring section appear to hold their breath, waiting.`
```

			`	Sure enough, the many minutes of silence are disrupted by echoing, increasingly louder screaming - one scream, incessant, coming from the shut passageway watched over by the sights of what must be a hundred Heliarch guns. A thud at the shut door briefly peaks out amidst the continuous shout. It opens to a lone, frantic Saryd, with a deeply perturbed look in his eyes. He doesn't stop at the sight of the several gun barrels pointed at him, instead drawing his own weapon, only to point it back to where he came from, firing wildly at the dark passage.`

> 			`	Then it stops. The screaming, and the gunshots, both stop abruptly, with the Saryd dropping the gun, then following it to the ground. He lays there, unresponsive, catatonic, as the Heliarch agents head closer to inspect him, eventually blocking him out entirely from view. A few seconds later, some of the Heliarchs back up, then bolt to a comms room. Others stand firmly on guard, with their guns pointed either at the Saryd, or at the passageway from which he came.`

Author: Changed it so it says restricted room instead.

> 			`	Then it stops. The screaming, and the gunshots, both stop abruptly, with the Saryd dropping the gun, then following it to the ground. He lays there, unresponsive, catatonic, as the Heliarch agents head closer to inspect him, eventually blocking him out entirely from view. A few seconds later, some of the Heliarchs back up, then bolt to a restricted room. Others stand firmly on guard, with their guns pointed either at the Saryd, or at the passageway from which he came.`

Reviewer 6: ```suggestion
			`	Then it stops. The screaming, and the gunshots, both stop abruptly, and the Saryd drops the gun, then collapses to the ground. He lays there, unresponsive, catatonic, as the Heliarch agents head closer to inspect him, eventually blocking him out entirely from view. A few seconds later, some of the Heliarchs back up, then bolt to a restricted room. Others stand firmly on guard, with their guns pointed either at the Saryd, or at the passageway from which he came.`
```

			branch heliarch
				has "joined the heliarchs"
			branch lunarium
				has "joined the lunarium"
			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet. They pay you no mind, and much to your relief steer a safe distance from your ship this time.`
				decline
			label heliarch
			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet, steering a safe distance from your ship this time. The captain of one of the ships, a Heliarch Interdictor, hails you, apologizing for the earlier distress, briefly mentioning they've gotten the situation back in control. They wish you a good day, and prepare to leave the system.`
				decline
			label lunarium

> 			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet. They pay you no mind, and much to your relief steer a safe distance from your ship this time. You cannot help but wonder whether the ships they were chasing held any Lunarium members, or if they were simply more unfortunate civilians.`

Reviewer: ```suggestion
			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet. They pay you no mind, and much to your relief steer to a safe distance from your ship this time. You cannot help but wonder whether the ships they were chasing held any Lunarium members, or if they were simply more unfortunate civilians.`
```

Reviewer: ```suggestion
			`	Some more minutes later, you see a select few of the Heliarch ships that passed close to you departing the planet. They pay you no mind, and much to your relief steer a safe distance away from your ship this time. You cannot help but wonder whether the ships they were chasing held any Lunarium members, or if they were simply more unfortunate civilians.`
```
Alternative idea.

Reviewer: (Honestly, I'm not 100% sure this is needed at all.)

Author: I don't really mind either way, so just tell me when you've decided on one of the changes or no changes to this.

Reviewer 4: I prefer the second one.

Reviewer: Let's go with the second one, then.

			`	They appear to be boarding the few ships that weren't damaged enough to have their hulls ruptured. After a few short minutes, they seem to have seized control of the vessels, and are having them follow the warships out of the system.`
			branch heliarch
				has "joined the heliarchs"
			branch lunarium
				has "joined the lunarium"
			branch neither
				not "joined the heliarchs"
				not "joined the lunarium"
			label heliarch
			`	One of the ships hails you briefly, apparently in a cheery mood as they celebrate having captured the criminals. They wish you a good day and good luck on your duties, and cut the comms as they prepare to leave.`
				decline
			label lunarium
			`	As you glance at the wreckage of the civilian ships, you feel uneasy. Worse still is the thought that hits you soon after, of whether any of the Lunarium leaders you met were aboard any of the ships.`
				decline
			label neither

> 				decline

Reviewer 5: Does this work?

Author: It seemed to end it properly when I tested it, yeah.
